src/main/webapp/app/exercises/shared/result/result.service.ts

```diff
@@ -31,12 +37,78 @@ export class ResultService implements IResultService {
     private resultResourceUrl = SERVER_API_URL + 'api/results';
     private participationResourceUrl = SERVER_API_URL + 'api/participations';
 
-    constructor(private http: HttpClient) {}
+    private readonly maxValueProgrammingResultInts = 255; // Size of tinyInt in SQL, that is used to store these values
+
+    constructor(private http: HttpClient, private translateService: TranslateService) {}
 
     find(resultId: number): Observable<EntityResponseType> {
         return this.http.get<Result>(`${this.resultResourceUrl}/${resultId}`, { observe: 'response' }).pipe(map((res: EntityResponseType) => this.convertDateFromServer(res)));
     }
 
+    getResultString(result?: Result, exercise?: Exercise): string {
+        if (result && exercise) {
+            return this.getResultStringNotUndefined(result, exercise);
+        } else {
+            captureException('Tried to generate a result string, but either the result or exercise was undefined');
+            return '';
+        }
+    }
+
+    private getResultStringNotUndefined(result: Result, exercise: Exercise): string {
```

Comment:
The method name does not sound good, please find a better one

Reply:
please also add some documentation to the method to make it easier to understand

Reply:
The method is wrapped by the method above called 'getResultString' which also allows undefined values for better error handling.
Sadly I did not find a way to overload the method and keep the name.
The name I now came up with is 'getResultStringDefinedParameters'. And I added comments as well